service/create/launch_configuration.go

```diff
@@ -102,7 +101,7 @@ func (s *Service) createLaunchConfiguration(input launchConfigurationInput) (boo
 		KeyName:                  input.keypairName,
 		SecurityGroupID:          securityGroupID,
 		SmallCloudConfig:         smallCloudconfig,
-		AssociatePublicIpAddress: publicIP,
+		AssociatePublicIpAddress: input.associatePublicIP,
```

Comment:
This is now configurable and only enabled for new clusters with the version set.